data/src/test/java/org/apache/iceberg/data/DataTestHelpers.java

```diff
@@ -85,14 +84,14 @@ private static void assertEquals(Type type, Object expected, Object actual) {
       case UUID:
       case BINARY:
       case DECIMAL:
-        Assert.assertEquals(
-            "Primitive value should be equal to expected for type " + type, expected, actual);
+        assertThat(actual)
+            .as("Primitive value should be equal to expected for type " + type)
+            .isEqualTo(expected);
         break;
       case FIXED:
         assertThat(expected).as("Expected should be a byte[]").isInstanceOf(byte[].class);
         assertThat(expected).as("Actual should be a byte[]").isInstanceOf(byte[].class);
-        Assert.assertArrayEquals(
-            "Array contents should be equal", (byte[]) expected, (byte[]) actual);
+        assertThat((byte[]) actual).as("Array contents should be equal").isEqualTo(expected);
```

Comment:
is the casting still necessary here? I believe this should also work without having to cast

Reply:
nope, removed, still looks good